src/compiler/manipulate_wasm_binary.ts

```diff
@@ -0,0 +1,244 @@
+// import { func, indexLiteral, instr } from '@webassemblyjs/ast';
+// import { addExport } from '@webassemblyjs/wasm-edit';
+// import { decode, encode } from '@webassemblyjs/wasm-parser';
+import binaryen from 'binaryen';
+import { readFile, writeFile } from 'fs/promises';
+import { join } from 'path';
+
+import { getConsumer } from './get_consumer_config';
+import { AZLE_PACKAGE_PATH } from './utils/global_paths';
+import { CanisterConfig, CompilerInfo } from './utils/types';
+
+// TODO put the licenses in the binary? Or with Azle? Probably with Azle actually
+// TODO it would be neat to be the licenses in all Azle binaries though
+// TODO can we make the start function just load the passive segment into memory?
+
+export async function manipulateWasmBinary(
+    canisterName: string,
+    js: string,
+    compilerInfo: CompilerInfo,
+    canisterConfig: CanisterConfig
+): Promise<void> {
+    const originalWasm = await readFile(
+        join(AZLE_PACKAGE_PATH, `static_canister_template.wasm`)
+    );
+
+    const module = binaryen.readBinary(originalWasm);
+
+    compilerInfo.canister_methods.queries.forEach(
+        ({ name: functionName, index, composite }) => {
+            addCanisterMethod(
+                module,
+                `${
+                    composite === true
+                        ? 'canister_composite_query'
+                        : 'canister_query'
+                } ${functionName}`,
+                'execute_js',
+                index,
+                true
+            );
+        }
+    );
+
+    compilerInfo.canister_methods.updates.forEach(
+        ({ name: functionName, index }) => {
+            addCanisterMethod(
+                module,
+                `canister_update ${functionName}`,
+                'execute_js',
+                index,
+                true
+            );
+        }
+    );
+
+    addCanisterMethod(
+        module,
+        'canister_init',
+        'init',
+        compilerInfo.canister_methods.init?.index ?? -1,
```

Comment:
So we add these methods onto all of the canisters and then if they are called and that index is -1 then we just don't execute_js. Is that right?